conda_build_all/tests/integration/test_builder.py

```diff
@@ -58,9 +60,15 @@ def test_no_source(self):
             r = builder.build(pkg1_resolved, conda_build.api.Config())
         else:
             r = builder.build(pkg1_resolved, conda_build.config.config)
-        self.assertTrue(os.path.exists(r))
-        self.assertEqual(os.path.abspath(r), r)
-        self.assertEqual(os.path.basename(r), 'pkg1-1.0-0.tar.bz2')
+        if isinstance(r, string_types):
+            rs = [r]
+        else:
+            rs = r
+        for r in rs:
+            self.assertTrue(os.path.exists(r))
+            self.assertEqual(os.path.abspath(r), r)
+            self.assertTrue(bool(re.match('pkg1-1.0-(h[0-9a-f]{7}_)?0.tar.bz2',
+                                          os.path.basename(r))))
```

Comment:
Because of handling the package hash I'm guessing? Seems fine.

Reply:
yes, exactly.  Allow it to be there (cb3), but it's ok too if it isn't (cb2)

Reply:
For sake of readability, might be worth pulling out the ``package hash`` concept into a variable and pulling it in in these asserts.